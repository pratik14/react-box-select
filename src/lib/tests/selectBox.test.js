import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import SelectBox from '../selectBox';

describe('<SelectBox />', () => {
  let options;

  beforeEach(() => {
    let step;
    options = [];
    for (step = 1; step < 10; step += 1) {
      options.push({ id: `${step}`, label: `Element ${step}` });
    }
  });

  it('should render without crashing', () => {
    const tree = renderer
      .create(
        <SelectBox
          type="selectable"
          options={options}
          selectors={{
            id: 'id',
            label: 'label',
          }}
          handleClick={() => {}}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('#handleOptionClick', () => {
    const opt = options[0];
    const mockCallback = jest.fn();
    const wrapper = shallow(
      <SelectBox
        type="selectable"
        options={options}
        selectors={{
          id: 'id',
          label: 'label',
        }}
        handleClick={mockCallback}
      />,
    );

    wrapper.instance().handleOptionClick(opt, 'selectable');
    expect(mockCallback.mock.calls.length).toBe(1);
    expect(mockCallback).toBeCalledWith(opt, 'selectable');
  });
});
