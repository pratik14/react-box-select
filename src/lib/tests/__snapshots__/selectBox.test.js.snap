// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectBox /> should render without crashing 1`] = `
<div
  className="ms-selectable"
>
  <ul
    className="ms-list"
  >
    <li
      className="ms-elem-selectable"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
    >
      Element 1
       
    </li>
    <li
      className="ms-elem-selectable"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
    >
      Element 2
       
    </li>
    <li
      className="ms-elem-selectable"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
    >
      Element 3
       
    </li>
    <li
      className="ms-elem-selectable"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
    >
      Element 4
       
    </li>
    <li
      className="ms-elem-selectable"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
    >
      Element 5
       
    </li>
    <li
      className="ms-elem-selectable"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
    >
      Element 6
       
    </li>
    <li
      className="ms-elem-selectable"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
    >
      Element 7
       
    </li>
    <li
      className="ms-elem-selectable"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
    >
      Element 8
       
    </li>
    <li
      className="ms-elem-selectable"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
    >
      Element 9
       
    </li>
  </ul>
</div>
`;
